Мария Лесничая, Suffix Tree

MariaForester opened the pull request.

MariaForester committed:
Node draft

MariaForester committed:
I`m developing my tree :3

MariaForester committed:
modules have been fixed; access is changed

MariaForester committed:
input string verification added

MariaForester committed:
boldness verification

MariaForester committed:
minor mixes, node number is added

MariaForester committed:
search function is added, some more minor fixes

MariaForester committed:
travis file is added

MariaForester committed:
tests

MariaForester committed:
travis test

MariaForester committed:
travis test

MariaForester committed:
travis test

MariaForester committed:
travis test

MariaForester committed:
travis test

MariaForester committed:
travis test

MariaForester committed:
travis test

MariaForester committed:
travis test

MariaForester committed:
tests  are fixed

DaniilPavlov reviewed (changes requested):
https://github.com/h31/ProgrammingLabAutumn2018/blob/a4c52fac1b79ef63c060e3a023366040af4f2f2d/src/main/java/SuffixTreeImplementation.java#L68-L75
Очень хорошая идея, мне нравится.

DaniilPavlov reviewed (commented):
Артем Олегович, проект я проверил. Все очень понравилось, даже придраться не к чему. По-моему это оценка *отлично*!

MariaForester commented:
- подача строки без доллара
- в draw ссылка прямая на Node, вместо подачи всего листа и извлечения узла по зиндексу
- добавить больше тестов на различные случаи для лучшей проверки
- убрать исключение, связанное с символом доллара
- изменить SuffixTreeImplementation на SuffixTree
- Tree is bold - так не должно быть, как есть сейчас
- Индекс. не Number
- Node class - инициализацию полей перенести в начало
- Добавление саффикса - разбить на есколько методов для компактности
- itHasSuchSUffix  - должно проверять в строке, есть ли в ней суффикс. идти в глубь дерева

MariaForester committed:
minor fixes

MariaForester committed:
more minor fixes

MariaForester committed:
major fixes (suffix search added)

MariaForester committed:
more fixes

MariaForester committed:
more fixes

MariaForester committed:
Calling a node itself instead of its index
No calling for "currentChildren" in search method
Getting rid of those younger children
other minor fixes

MariaForester committed:
variables are renamed. so proud of myself

MariaForester committed:
search for substring

MariaForester committed:
last method is fixed, more tests are added

MariaForester committed:
more fixes, more tests

MariaForester committed:
tests division

MariaForester committed:
...

MariaForester committed:
indices -> index in Node class

MariaForester committed:
Add files via upload

h31 commented:
Принято.
